### ISSUE TITLE
Trace Context Propagation with HTTP instrumentation produce the invalid RootSpan

### ISSUE THREAD

**Issue by yamadayuki** [census-instrumentation/opencensus-node#546]
### What version of OpenCensus are you using?

I use the OpenCensus libraries which versions are 0.0.13.

- `@opencensus/core@0.0.13`
- `@opencensus/exporter-stackdriver@0.0.13`
- `@opencensus/instrumentation-http@0.0.13`
- `@opencensus/nodejs@0.0.13`
- `@opencensus/propagation-tracecontext@0.0.13`

### What version of Node are you using?

Node v10.15.3 (on macOS Mojave 10.14.4)

```
$ node -v 
v10.15.3
```

### What did you do?

I have written a simple Express app with OpenCensus.

```ts
import { Tracer } from "@opencensus/core";
import { StackdriverTraceExporter } from "@opencensus/exporter-stackdriver";
import * as instrumentationHttp from "@opencensus/instrumentation-http";
import tracing from "@opencensus/nodejs";
import { TraceContextFormat } from "@opencensus/propagation-tracecontext";
import express from 'express';
import * as http from 'http';

const propagation = new TraceContextFormat();
const exporter = new StackdriverTraceExporter({ projectId: "project-id" });

const tracer = tracing.start({ propagation, exporter }).tracer;
instrumentationHttp.plugin.enable(http, tracer as Tracer, process.versions.node, {});

const app = express();

app.get('/ping', (_req, res) =>  {
  res.send('pong');
});

app.listen(3000, () => {
  console.log("App listening on 3000");
});
```

When I request the `curl http://localhost:3000/ping` and then the Stackdriver exporter doesn’t export any trace. The error told me the RootSpan's TraceID is invalid.

### What did you expect to see?

I want to use `instrumentation-http` with `propagation-tracecontext`. The TraceID should be a valid string.

### What did you see instead?

HTTP instrumentation produces the invalid Span when I use with Trace Context propagation. This “invalid” means the TraceID is to be an empty string. But the TraceID should be hexadecimal 32-bits string. So the exporter raises the error and the exporter does not export.

### Additional context

I tried to use HTTP instrumentation with B3 Propagation and Stackdriver Propagation. Both B3 and Stackdriver propagation and HTTP instrumentation combination create proper Span.
So this problem may be caused by using HTTP instrumentation and Trace Context Propagation.

**Comment by yamadayuki** [comment]
I investigated this problem, I supposed that I found the cause of this problem.

In `http.ts`, the `spanContext` from `propagation.extract` is weird when I use the Trace Context Propagation.
```
if (propagation) {
  const spanContext = propagation.extract(getter);
  console.log({spanContext}); // I added for debug
  if (spanContext) {
    traceOptions.spanContext = spanContext;
  }
}
```

Output is below:

```js
{ spanContext:
   { traceId: '', spanId: '', options: 0, traceState: undefined } }
```

But the `spanContext` with B3 Propagation is correct.

```js
{ spanContext: null }
```

I think this problem occurs only at the beginning of the tracing (under the context with no TraceParent).

**Comment by mayurkale22** [comment]
@yamadayuki Thanks for reporting this! I am looking into it.

**Comment by mayurkale22** [comment]
I think, the `extract` method should [return `null`](https://github.com/census-instrumentation/opencensus-node/blob/3111ab8c29b5d60e68a762c439ef909e3af9589b/packages/opencensus-core/src/trace/propagation/types.ts#L37) if there is no incoming `traceID`/`spanID`, rather than fill `SpanContext` object [with empty strings](https://github.com/census-instrumentation/opencensus-node/blob/master/packages/opencensus-propagation-tracecontext/src/tracecontext-format.ts#L46-L52). I will open a PR to fix this.

**Comment by yamadayuki** [comment]
Thanks for triage! I agree with your comment https://github.com/census-instrumentation/opencensus-node/issues/546#issuecomment-495340712. I'm looking forward to the next release.

**Comment by mayurkale22** [comment]
@yamadayuki The new release is available (0.0.14) with this fix.
